File: src/api/controllers/artwork_controller.py
from fastapi import HTTPException
from fastapi.responses import FileResponse
from pathlib import Path


async def get_series_backdrop(series_id: str):
    file_path = Path(f"config/artwork/series/{series_id}/backdrop.webp")
    if not file_path.exists() or file_path.is_dir():
        raise HTTPException(status_code=404, detail="Backdrop not found")
    return FileResponse(str(file_path), media_type="image/webp")


async def get_series_poster(series_id: str):
    file_path = Path(f"config/artwork/series/{series_id}/poster.webp")
    if not file_path.exists() or file_path.is_dir():
        raise HTTPException(status_code=404, detail="Poster not found")
    return FileResponse(str(file_path), media_type="image/webp")

File: internal/api/controllers/artwork_controller.go
package controllers

import (
	"net/http"
	"os"
	"path/filepath"
	"transfigurr/internal/config"
)

type ArtworkController struct{}

func NewArtworkController() *ArtworkController {
	return &ArtworkController{}
}

func (a *ArtworkController) GetSeriesBackdrop(w http.ResponseWriter, r *http.Request, seriesId string) {
	filePath := filepath.Join(config.ArtworkPath, "series", seriesId, "backdrop.webp")
	if _, err := os.Stat(filePath); os.IsNotExist(err) {
		http.Error(w, "Backdrop not found", http.StatusNotFound)
		return
	}
	http.ServeFile(w, r, filePath)
}

func (a *ArtworkController) GetSeriesPoster(w http.ResponseWriter, r *http.Request, seriesId string) {
	filePath := filepath.Join(config.ArtworkPath, "series", seriesId, "poster.webp")
	if _, err := os.Stat(filePath); os.IsNotExist(err) {
		http.Error(w, "Poster not found", http.StatusNotFound)
		return
	}
	http.ServeFile(w, r, filePath)
}

func (a *ArtworkController) GetMovieBackdrop(w http.ResponseWriter, r *http.Request, movieId string) {
	filePath := filepath.Join(config.ArtworkPath, "movies", movieId, "backdrop.webp")
	if _, err := os.Stat(filePath); os.IsNotExist(err) {
		http.Error(w, "Backdrop not found", http.StatusNotFound)
		return
	}
	http.ServeFile(w, r, filePath)
}

func (a *ArtworkController) GetMoviePoster(w http.ResponseWriter, r *http.Request, movieId string) {
	filePath := filepath.Join(config.ArtworkPath, "movies", movieId, "poster.webp")

	if _, err := os.Stat(filePath); err != nil {
		if os.IsNotExist(err) {
			http.Error(w, "Poster not found", http.StatusNotFound)
			return
		}
		// Handle other potential errors
		http.Error(w, "Error accessing poster", http.StatusInternalServerError)
		return
	}

	http.ServeFile(w, r, filePath)
}
